### PR TITLE
Move localizing code from SipHash to a general helper

### PR DESCRIPTION
**Pull request by ronawho**
In #287 SipHash was optimized by localizing segmented string data with a
bulk GET instead of collecting elements one at a time. This moves that
localization logic into a common `lowLevelLocalizingSlice` helper so
that we can use it in an upcoming PR to localize segmented string data
used in the regex implementation.

This also adds a seed argument to SipHashSpeedTest to get the same
arrays from run-to-run. I used this to verify there is no performance
difference from this change.

Part of #917

### CODE REVIEW

**Review by reuster986 — APPROVED** [review]
cool!

**Comment by reuster986** [comment]
I'll wait for @pierce314159 to review before merging, since he will be using this code directly.

**Review by pierce314159 — APPROVED** [review]
Looks good to me! Thanks @ronawho!

**Pull request merged**
